CHANGELOG.md

```diff
@@ -5,6 +5,9 @@
   * This allows you to specify a json-schema for the schema files themselves, providing hints and validation in your editor
 * `exh data schemas sync` now allows all components in a schema to have a `description` property
   * These descriptions are not synced, but allow you to provide inline documentation for the components in your schema
+* Revamped the schema validation to use the more complete json-schema definition
+  * This affects `exh data schemas verify` as well as `exh data schemas sync` and more accurately reports errors in your schemas
+  * The `--ignoreSchemaVerificationErrors` flag can always be used to sync while you might be working on a schema with errors
```

Comment:
I've tested the validation extensively with all existing schemas and synced the `tests/__helpers__/schemas/fairlyComplete.json
` listed below, which should cover every option we support in the data service.

I'm quite certain it is best to even throw new errors for users upgrading to this version, as their definition would be broken if it doesn't pass this validation. Therefore I kept the strict validation, not as we had discussed last time with Aäron, but I'm willing to take the heat for it if any user runs into issues.

That's also the reason why I included a reminder about the `--ignoreSchemaVerificationErrors` flag, as this should always unblock any user running into errors in the validation.

**Happy to discuss!**

Reply:
I'm not a huge fan of two statements in this message:

- I'm quite certain it is best to even throw new errors for users upgrading to this version
I'm still quite worried that (to use your own word) the 'heat' we've gotten over the CLI's validation issues won't result in significant backlash if 'new things' are broken after the 'validation revamp' but I guess this can be solved with the correct expectations being set with our clients?

-  but I'm willing to take the heat for it if any user runs into issues.
You're not a solo dev, we ride or die together fam'. I trust that you trust the heat is worth it and we're all responsible.

Reply:
* Fair, I will adjust the message to motivate the reasoning an upsides and proactively ask for feedback. Within this changelog, but when sending the message out to who we want to address specifically?
* Also fair, it maybe was also more meant like "I'm willing to take the risk and go out of my way to fix it (quickly)", rather then having the validation not catch errors with (unintentional) unknown fields